pkg/handlers/v1/nexposeassetproducer.go

```diff
@@ -66,6 +66,13 @@ func (h *NexposeScannedAssetProducer) Handle(ctx context.Context, in ScanInfo) {
 				errChan = nil
 			} else {
 				switch err.(type) {
+				// in the event of an ErrorFetchingAssets case, we are guaranteed that we have not produced ANY assets.
```

Comment:
Task: The handler logic needs to be reworked, unfortunately, to match the interface change requested above. There is no need for a case switch here anymore. Just receive all the assets, and either fail on an error, or produce them all concurrently.

Reply:
done